### ISSUE TITLE
Accessibility: Edit folder button has no text

### ISSUE THREAD

**Issue by ChloeSartorelli** [silverstripe/silverstripe-asset-admin#1597]
The button for editing a folder (seen in the breadcrumbs of a folder, example below) has no text:

<img width="381" height="125" alt="Image" src="https://github.com/user-attachments/assets/7aa96067-c35a-4c00-815f-c22554b14a55" />

For screen readers this means there is a button with effectively no label (technically the screen reader will see a colon because of the icon implementation, but that's being fixed in https://github.com/silverstripe/silverstripe-admin/issues/1957). For sighted users there is a button with an icon that may-or-may-not make sense to them.

## Acceptance Criteria

- [ ] The button has a label that will be read out loud by a screen reader
- [ ] The button has a label that will be displayed on hover for sighted readers

**Do not merge until CMS 6.1 beta has been released**

## PRs
- https://github.com/silverstripe/silverstripe-admin/pull/1977
- https://github.com/silverstripe/silverstripe-asset-admin/pull/1598

**Comment by ChloeSartorelli** [comment]
PRs merged
